src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -132,8 +132,32 @@ async def _run(self) -> None:  # pylint: disable=too-many-locals
         """
         await self._component_manager.start()
 
+        power_tasks: dict[str, asyncio.Task[None]] = {}
+
+        def cleanup(task: asyncio.Task[None]) -> None:
+            task_name = task.get_name()
+            assert task_name in power_tasks, "Task not found in power_tasks"
+
+            # Raise if there was an exception while processing the task
+            task.result()
+
+            del power_tasks[task_name]
+
         async for request in self._requests_receiver:
-            await self._component_manager.distribute_power(request)
+            task_name = f"power_request_ids={request.component_ids}"
```

Comment:
I updated to use a frozenset but I still need the task name for lookups once a task processed/done

Reply:
What do you think about computing the hash of a frozenset as they are hasheable? The main disadvantages I can think of are it won't be helpful information for debugging and it isn't reproducible in different platforms/environments. But it should be cheaper the conversion plus sorting and it would be a relatively short string

Reply:
I've just sorted it at the moment which converts it to a sorted list

Reply:
okay, using the hash sounds like a good idea as well.  But I'll let you decide.